app/services/ipam/base.rb

```diff
@@ -41,6 +41,18 @@ def serializable_hash(options = nil)
 
     private
 
+    def block_ip_cache_key(ip)
+      "blocked_ip_cache_#{subnet.id}/#{ip}"
+    end
+
+    def block_ip(ip)
+      Rails.cache.write(block_ip_cache_key(ip), 'blocked', expires_in: 30.minutes)
```

Comment:
30 should at least be a constant and all places that now use 30 should refer to it. If we ever want to change it, it's easy.

Reply:
yes. thanks!